tuplex/codegen/src/BlockGeneratorVisitor.cc

```diff
@@ -1659,6 +1659,19 @@ namespace tuplex {
                     keyval.second = std::vector<python::Type>{python::Type::UNKNOWN}; // save result
                 }
 
+                if(keyval.second.front().isIteratorType()) {
+                    VariableSlot slot;
```

Comment:
ok, some comment within this if statement on what happens would be highly appreciated. Also, what about the continue?

Reply:
Since python iteratorType to llvm iterator type is a one-to-many mapping, we set slot.var.ptr of an iterator to nullptr during declareVariables(), and fill in concrete llvm type later in assignToSingleVariable(). 
Refactored and added comments.